pooch/utils.py

```diff
@@ -59,20 +61,22 @@ def file_hash(fname):
     >>> with open(fname, "w") as f:
     ...     __ = f.write("content of the file")
     >>> print(file_hash(fname))
-    0fc74468e6a9a829f103d069aeb2bb4f8646bad58bf146bb0e3379b759ec4a00
+    sha256:0fc74468e6a9a829f103d069aeb2bb4f8646bad58bf146bb0e3379b759ec4a00
     >>> import os
     >>> os.remove(fname)
 
     """
     # Calculate the hash in chunks to avoid overloading the memory
     chunksize = 65536
-    hasher = hashlib.sha256()
+    if alg not in hashlib.algorithms_available:
+        raise ValueError("Algorithm '{}' not available in hashlib".format(alg))
+    hasher = hashlib.new(alg)
     with open(fname, "rb") as fin:
         buff = fin.read(chunksize)
         while buff:
             hasher.update(buff)
             buff = fin.read(chunksize)
-    return hasher.hexdigest()
+    return "{}:{}".format(alg, hasher.hexdigest())
```

Comment:
This is the only part that I'm not very sure what to do. This technically breaks compatibility since `file_hash` is in the public API. If possible, I'd prefer to keep this function returning only the hash to be on the safe side.

Reply:
Fixed in 0a5e712